homeassistant/components/ambient_station/__init__.py

```diff
@@ -41,266 +20,43 @@
-    TYPE_SOILTEMP6F: ("Soil Temp 6", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_SOILTEMP7F: ("Soil Temp 7", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_SOILTEMP8F: ("Soil Temp 8", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_SOILTEMP9F: ("Soil Temp 9", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_SOLARRADIATION: (
-        "Solar Rad",
-        IRRADIATION_WATTS_PER_SQUARE_METER,
-        SENSOR,
-        None,
-    ),
-    TYPE_SOLARRADIATION_LX: (
-        "Solar Rad (lx)",
-        LIGHT_LUX,
-        SENSOR,
-        DEVICE_CLASS_ILLUMINANCE,
-    ),
-    TYPE_TEMP10F: ("Temp 10", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_TEMP1F: ("Temp 1", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_TEMP2F: ("Temp 2", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_TEMP3F: ("Temp 3", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_TEMP4F: ("Temp 4", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_TEMP5F: ("Temp 5", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_TEMP6F: ("Temp 6", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_TEMP7F: ("Temp 7", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_TEMP8F: ("Temp 8", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_TEMP9F: ("Temp 9", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_TEMPF: ("Temp", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_TEMPINF: ("Inside Temp", TEMP_FAHRENHEIT, SENSOR, DEVICE_CLASS_TEMPERATURE),
-    TYPE_TOTALRAININ: ("Lifetime Rain", PRECIPITATION_INCHES, SENSOR, None),
-    TYPE_UV: ("uv", "Index", SENSOR, None),
-    TYPE_WEEKLYRAININ: ("Weekly Rain", PRECIPITATION_INCHES, SENSOR, None),
-    TYPE_WINDDIR: ("Wind Dir", DEGREE, SENSOR, None),
-    TYPE_WINDDIR_AVG10M: ("Wind Dir Avg 10m", DEGREE, SENSOR, None),
-    TYPE_WINDDIR_AVG2M: ("Wind Dir Avg 2m", SPEED_MILES_PER_HOUR, SENSOR, None),
-    TYPE_WINDGUSTDIR: ("Gust Dir", DEGREE, SENSOR, None),
-    TYPE_WINDGUSTMPH: ("Wind Gust", SPEED_MILES_PER_HOUR, SENSOR, None),
-    TYPE_WINDSPDMPH_AVG10M: ("Wind Avg 10m", SPEED_MILES_PER_HOUR, SENSOR, None),
-    TYPE_WINDSPDMPH_AVG2M: ("Wind Avg 2m", SPEED_MILES_PER_HOUR, SENSOR, None),
-    TYPE_WINDSPEEDMPH: ("Wind Speed", SPEED_MILES_PER_HOUR, SENSOR, None),
-    TYPE_YEARLYRAININ: ("Yearly Rain", PRECIPITATION_INCHES, SENSOR, None),
-}
-
 CONFIG_SCHEMA = cv.deprecated(DOMAIN)
 
 
+@callback
+def async_wm2_to_lx(value: float) -> int:
+    """Calculate illuminance (in lux)."""
+    return round(value / 0.0079)
+
+
+@callback
+def async_hydrate_station_data(data: dict[str, Any]) -> dict[str, Any]:
+    """Hydrate station data with addition or normalized data."""
+    if TYPE_SOLARRADIATION in data:
+        data[TYPE_SOLARRADIATION_LX] = async_wm2_to_lx(data[TYPE_SOLARRADIATION])
```

Comment:
```suggestion
    if (irradiation := data.get(TYPE_SOLARRADIATION)) is not None:
        data[TYPE_SOLARRADIATION_LX] = async_wm2_to_lx(irradiation)
```
Something like this should work. What do you think?

Reply:
It should, but why? Is this functionally better than what's already there? We still end up with two lines of code, but I'd argue that this is unnecessarily complicated.

Reply:
The suggestion only uses one dictionary lookup compared to two